### PR TITLE
Error on adding free shipping voucher to draft order

### PR DESCRIPTION
**Pull request by CossackDex**
I want to merge this change because it fixes error on adding free shipping voucher to draft order.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4138 Error on adding free shipping voucher](https://app.clickup.com/t/2549495/SALEOR-4138)

File: saleor/graphql/order/tests/test_order.py [hunk 1]
```diff
@@ -2591,6 +2591,51 @@ def test_draft_order_update_doing_nothing_generates_no_events(
...
+    }
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_orders]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["draftOrderUpdate"]
+    assert not data["errors"]
+    assert data["order"]["id"] == variables["id"]
+    order.refresh_from_db()
+    assert order.voucher
```
Comment (L3str4nge) [inline saleor/graphql/order/tests/test_order.py:2636]: Consider checking `voucher id` in assertion.

File: saleor/order/utils.py [hunk 2]
```diff
@@ -133,6 +133,8 @@ def recalculate_order_prices(order: Order, **kwargs):
     voucher_discount = kwargs.get("discount", zero_money(order.currency))
 
     # discount amount can't be greater than order total
+    if type(voucher_discount) == TaxedMoney:
```
Comment (L3str4nge) [inline saleor/order/utils.py:136]: Wouldn't be better to call `isinstance`?

File: saleor/discount/models.py [hunk 3]
```diff
@@ -119,6 +119,8 @@ def get_discount(self, channel: Channel):
     def get_discount_amount_for(self, price: Money, channel: Channel):
         discount = self.get_discount(channel)
         after_discount = discount(price)
+        if type(after_discount) is TaxedMoney:
```
Comment (korycins) [inline saleor/discount/models.py:122]: See typing for this method. If you're receiving here TaxedMoney, I would say that there is an incorrect call of this method as typing expects to get normal Money price.

File: saleor/order/utils.py (hunk 2 above)
Comment (korycins) [reply saleor/order/utils.py:136]: As in the first `if` condition. If you're receiving here voucher_discount as a TaxedMoney, I would say that we should fix the place when this value is set to TaxedMoney, not by converting them to gross (which is also not ideal behaviour as .gross/.net should depend here on Saleor's setting `include_taxes_in_prices`)

**Review by korycins — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**